generate-release-notes.py

```diff
@@ -28,14 +47,22 @@ def issues_in_milestone():
         if m.title == MILESTONE:
             # Found the milestone in this repo - look for issues (but only
             # ones that have been closed!)
+            print(f"  found milestone {m.title}")
             issues = []
-            for i in repo.get_issues(milestone=m, state="closed", labels=['release-note-required']):
-                pr = i.as_pull_request()
+            milestone_issues = repo.get_issues(
+                milestone=m, state="closed", labels=["release-note-required"]
+            )
+            for issue in milestone_issues:
+                pr = issue.as_pull_request()
                 if pr.merged:
                     # Filter out PRs which are closed but not merged.
-                    issues.append(i)
+                    issues.append(issue)
+                else:
+                    print(f"WARNING: {pr.number} is not merged, skipping")
+            if len(issues) == 0:
```

Comment:
This should just be `if issues:`

Also, I would move this above the loop for clarity's sake — weird to loop over something and then check if it's empty afterwards.

Reply:
you are conflating `milestone_issues` with `issues`. It will be a noop if before the loop meanwhile at this point is to see if there are issues to generate a release note for